### PR TITLE
22335: Allow separate gateway IAM role and policy for AWSGov and AWSChina and add IAM role based account for AWSGov

### PR DESCRIPTION
**Pull request by mlin-aviatrix**

### CODE REVIEW

File: docs/resources/aviatrix_account.md [hunk 1]
```diff
@@ -176,8 +176,11 @@ The following arguments are supported:
 
 ### AWSGov Cloud
 * `awsgov_account_number` - (Optional) AWSGov Account number to associate with Aviatrix account. Required when creating an account for AWSGov.
+* `awsgov_iam` - (Optional) AWSGov IAM-role based flag. Available as of provider version 2.19+.
 * `awsgov_access_key` - (Optional) AWS Access Key. Required when creating an account for AWSGov.
 * `awsgov_secret_key` - (Optional) AWS Secret Key. Required when creating an account for AWSGov.
+* `awsgov_role_app` - (Optional) AWSGov App role ARN. Available when `awsgov_iam` is "true" and when creating an account for AWSChina. If left empty, the ARN will be computed. Available as of provider version 2.19+.
+* `awschina_role_ec2` - (Optional) AWSGov EC2 role ARN. Available when `awsgov_iam` is "true" and when creating an account for AWSGov. If left empty, the ARN will be computed. Available as of provider version 2.19+.
```
Comment (inc1t3Ful) [inline docs/resources/aviatrix_account.md]: typo. should be awsgov

File: aviatrix/resource_aviatrix_account.go [hunk 2]
```diff
@@ -396,11 +428,20 @@ func resourceAviatrixAccountCreate(ctx context.Context, d *schema.ResourceData,
...
-			return diag.Errorf("aws gov access key needed for aws gov cloud")
-		}
-		if account.AwsgovSecretKey == "" {
-			return diag.Errorf("aws gov secret key needed for aws gov cloud")
+		if awsGovIam {
+			if account.AwsgovRoleApp == "" {
+				account.AwsgovRoleApp = fmt.Sprintf("arn:aws-us-gov:iam::%s:role/aviatrix-role-app", account.AwsChinaAccountNumber)
+			}
+			if account.AwsgovRoleEc2 == "" {
+				account.AwsgovRoleEc2 = fmt.Sprintf("arn:aws-us-gov:iam::%s:role/aviatrix-role-ec2", account.AwsChinaAccountNumber)
```
Comment (inc1t3Ful) [inline aviatrix/resource_aviatrix_account.go]: why are we setting AWS China account number for AWS Gov IAM?

Comment (mlin-aviatrix) [reply aviatrix/resource_aviatrix_account.go]: Its a typo as well

**Review by inc1t3Ful — DISMISSED** [review]

File: aviatrix/resource_aviatrix_account.go [hunk 3]
```diff
@@ -313,28 +332,41 @@ func resourceAviatrixAccountCreate(ctx context.Context, d *schema.ResourceData,
...
+	} else {
+		account.AwsgovIam = "false"
+	}
+
+	awsChinaIam := d.Get("awschina_iam").(bool)
+	if awsChinaIam {
+		account.AwsChinaIam = "true"
+	} else {
+		account.AwsChinaIam = "false"
+	}
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_account.go]: Use strconv.FormatBool

File: aviatrix/resource_aviatrix_account.go [hunk 4]
```diff
@@ -618,6 +674,13 @@ func resourceAviatrixAccountUpdate(ctx context.Context, d *schema.ResourceData,
 		account.AwsIam = "false"
 	}
 
+	awsGovIam := d.Get("awsgov_iam").(bool)
+	if awsGovIam {
+		account.AwsgovIam = "true"
+	} else {
+		account.AwsgovIam = "false"
+	}
```
Comment (CyrusJavan) [inline aviatrix/resource_aviatrix_account.go]: strconv.FormatBool

**Review by CyrusJavan — CHANGES_REQUESTED** [review]

**Review by zhiqiangjin — APPROVED** [review]

**Review by CyrusJavan — APPROVED** [review]

**Pull request merged**